### PR TITLE
Draft: Preview/1718 visual indicator for ta automation robot

### PR DESCRIPTION
**Pull request by ckr123**

### CODE REVIEW

File: domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementAutomationControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        var response = await authenticatedClient.GetAsync("transfer-automation/status");
+        response.EnsureSuccessStatusCode();
+
+        var status = JsonConvert.DeserializeObject<TransferAutomationStatus>(await response.Content.ReadAsStringAsync());
+
+        status.Should().BeEquivalentTo(new TransferAutomationStatus(CacheValues.Success));
+    }
+
+    [Fact]
+    public async Task GetStatus_ShouldReturnError()
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementAutomationControllerTest.cs:47]: WhenWhat?

File: domains/transfer/Transfer.API/API.IntegrationTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+public class TransferAgreementsAutomationServiceTest
+{
+    [Fact]
+    public async Task Run_WhenFail_CacheIsSetToError()
+    {
+        var logger = Substitute.For<ILogger<TransferAgreementsAutomationService>>();
+        var transferAgreementRepository = Substitute.For<ITransferAgreementRepository>();
+        var projectOriginWalletService = Substitute.For<IProjectOriginWalletService>();
+        var metrics = Substitute.For<ITransferAgreementAutomationMetrics>();
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API.IntegrationTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs:28]: It's not an integration test when everything i mocked

File: domains/transfer/Transfer.API/API/Controllers/TransferAgreementAutomationController.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    [ProducesResponseType(typeof(TransferAutomationStatus), 200)]
+    [HttpGet("transfer-automation/status")]
+    public async Task<ActionResult<TransferAutomationStatus>> GetStatus()
+    {
+        cache.Cache.TryGetValue(CacheValues.Key, out var value);
+
+        return Ok(value == null ?
+            new TransferAutomationStatus(CacheValues.Error) :
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API/Controllers/TransferAgreementAutomationController.cs:29]: Is it an error if the transfer robot has not run yet?

File: domains/transfer/Transfer.API/API.IntegrationTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+using API.TransferAgreementsAutomation;
+using FluentAssertions;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using Xunit;
+using Exception = System.Exception;
+
+namespace API.IntegrationTests.TransferAgreementsAutomation.Service;
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API.IntegrationTests/TransferAgreementsAutomation/Service/TransferAgreementsAutomationServiceTest.cs:18]: Did you consider writing actual integration tests, where everything is spun up?

File: domains/transfer/Transfer.API/API.IntegrationTests/SwaggerTests.GetSwaggerDoc_AppStarted_NoChangesAccordingToSnapshot.verified.txt [hunk 5]
```diff
@@ -671,6 +700,15 @@
           }
         },
         "additionalProperties": false
+      },
+      "TransferAutomationStatus": {
+        "type": "object",
+        "properties": {
+          "status": {
+            "type": "string"
```
Comment (duizer) [inline domains/transfer/Transfer.API/API.IntegrationTests/SwaggerTests.GetSwaggerDoc_AppStarted_NoChangesAccordingToSnapshot.verified.txt:708]: Why is the status a magic string? Why not use something more type safe e.g. a boolean value called "healthy"; when true it is healthy, when false it is not healty. 
Or potentially use http status codes for this

**Review by tnickelsen — APPROVED** [review]

**Pull request merged**